### ISSUE TITLE
Screen Reader Usability Fixes on AAC Form

### ISSUE THREAD

**Issue by mateoclarke** [cityofaustin/techstack#268]
RE: https://cityofaustin.github.io/prototypeform/

In our usability testing with Glenda, she was not able to fill out any of the "Add additional information" text areas. The button that triggers making those textareas visible didn't do anything. Nor did any of the conditionals that show warnings or extra information boxes. Also, the page that has you add existing animals didn't work either. It seems like her setup had javascript disabled but its hard to be sure.

She said she was using Windows 7, Internet Explorer 11,  Jaws 18 and Zoom Text 11. She also has Fusion but apparently wasn't using it.

Here are some potential solutions/things to check...
1. Check to see if the form works as expected in IE11 using BrowserStack
1. Add `no-js` class and display a warning message to users that Javascript is disabled in their browser and the form will not work 100% as designed. https://stackoverflow.com/questions/6724515/what-is-the-purpose-of-the-html-no-js-class
1. Change the styling of textareas from being `display: none` or whatever they are now to be visually hidden so they exist on the DOM for vision impaired users: https://css-tricks.com/places-its-tempting-to-use-display-none-but-dont/
1. Wondering if the way we add more stuff to the DOM when we add blocks for animal info is ok or if there is a better pattern for that too.

Ideally, we would troubleshoot this before testing with Sharon rescheduled for Friday 4/13.

Any other ideas of things we should do @ifsimicoded?

**Comment by mateoclarke** [comment]
@juliazbyron the progress bar text didn't display but the screen reader read it out. It read out at the beginning of each page, "you are on step x of y." Check the tape.

The way I interpreted what Glenda was saying about optional text areas is that they should always be accessible by screen readers instead of hidden behind a button which she said felt like the wrong use for a button. Glenda liked how the optional textareas worked on the last page and was confused she couldn't access them at all on the rest of the pages. I understand what you're saying, but I think it would be better to give it a shot and see if another user things its too cumbersome to have to skip through them.

**Comment by mateoclarke** [comment]
Update after doing the first check I listed above:
> 1. Check to see if the form works as expected in IE11 on Windows 7 using BrowserStack

Turns out we were are hitting a syntax error in our javascript code because we wrote the code using the current ES6 syntax. IE11 doesn't support modern javascript (ES6) 😞 . The best practice is to compile the code using a tool called [Babel](http://babeljs.io/) so it has backwards compatible with old browsers. I forgot to do this since most of the browsers we work with don't choke on ES6. (my bad)

To fix this, I manually ran the code through Babel at http://babeljs.io/repl/ and copy pasted the result into a new file called `browser-safe-index.js` and retested. Here is the commit where I did that: https://github.com/cityofaustin/prototypeform/commit/1daae1b1da1ae2c451d42a8dc8b2ab23d2eb1739

## Before (no warning 👿 )
![bs_win7_ie_11 0](https://user-images.githubusercontent.com/5697474/38655106-43d75242-3dd8-11e8-9171-060eb328c042.jpg)

## After  (warning appears!! 😇)

![bs_win7_ie_11 0 1](https://user-images.githubusercontent.com/5697474/38655110-53db0c7e-3dd8-11e8-8c4e-a123e963e2b8.jpg)

In the future, we'll want to add Babel to the project so that it automatically compiles the Javascript code into an old browser-safe version. We don't need to worry about this until (and if) we make further changes to the index.js file.

This should fix it! 🚀 cc/ @juliazbyron @ifsimicoded

**Comment by juliazbyron** [comment]
@mateoclarke re: your responses

Prog bar - I totally didn't hear it read out, I'm glad you heard it! I was trying to listen for them but I was trying to listen for a lot of things so I missed it. 

Display:none - Thanks for clarifying, I like that idea. I misinterpreted the note to be about all items with display:none (eg alerts), not just the "add additional info" text areas. Thanks for clarifying and taking care of it. I also don't think we'll use buttons for that function in the real thing because she made the good point that buttons usually take you somewhere else. I knew that when I decided to use the buttons for prototype purposes but I didn't realize the repercussions for screen readers. A good learning for a non-dev hacking something together.
